bip-vaults.mediawiki

```diff
@@ -0,0 +1,745 @@
+** If this value is not 32 bytes, the script MUST fail and terminate immediately.
+
+* <code><trigger-sPK></code> is a variable length data push.
+** It must be the scriptPubKey that is the preimage to the <code><trigger-sPK-hash></code> specified in the spent <code>OP_VAULT</code> input.
+** If this value does not tagged-hash to <code><trigger-sPK-hash></code> supplied by the <code>OP_VAULT</code> parameter, the script MUST fail and terminate immediately.
+*** Verify <code>tagged_hash("VaultTriggerSPK", <trigger-sPK>) == <trigger-sPK-hash></code>
+** If this value does not correspond to a valid witness program, the script MUST fail and terminate immediately.
+
+* the remaining elements serve as the witness stack to satisfy the <code><trigger-sPK></code> witness program.
+** If <code>VerifyWitnessProgram(<remaining stack elements>, <trigger-sPK>, ...)</code> fails, the script MUST fail and terminate immediately.
+** (This validates that the withdrawal trigger has been authorized.)
+
+===== Transaction outputs validation =====
+
+Once the contents of the witness stack have been parsed and validated, the transaction outputs must be checked.
+
+First, we must check for a ''revault output'': an output in the trigger transaction whose
+scriptPubKey exactly matches that of the <code>OP_VAULT</code> input being
+spent. Its presence is optional.
+
+For each vault input citing a particular <code><trigger-vout-idx></code>, the output 
+located at <code>vout[<trigger-vout-idx>]</code> (the "trigger output") must:
+
+* have as its scriptPubKey a witness program version 1 with a single <code>OP_UNVAULT</code> tapscript, having the internal key <code>lift_x(0x0250929b74c1a04954b78b4b6035e97a5e078a5a0f28ec96d547bfee9ace803ac0)</code>, per the NUMS point mentioned in [https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki#user-content-Design BIP-0341].<ref>'''Why must the OP_UNVAULT taproot use a NUMS point as its internal key?''' This ensures that an OP_UNVAULT trigger output is verifiable as expected. It also ensures that it is spendable only by the conditions of the vault.</ref>
+** If the witness program has a version less than 1, the script MUST fail and terminate immediately.
+** If the scriptPubKey of the output does not match the expected scriptPubKey, as computed by creating a taproot output using the cited NUMS point and a single tapscript spend condition of the form<br /><code><recovery-params-from-stack> <spend-delay-from-stack> <target-outputs-hash-from-stack> OP_UNVAULT</code>,<br />the script MUST fail and terminate immediately.
+** Witness versions greater than 1 are allowed for upgradeability.
```

Comment:
The parent bullet point requires version 1, so which is it?

Reply:
I've clarified this in ca12c526ab11b0e7bf2b2da693405cf8ac350bfa.